tests/integration/test_install_uri.py

```diff
@@ -92,3 +92,32 @@ def test_install_editable_git_tag(PipenvInstance, pip_src_dir, pypi):
         assert 'git' in p.lockfile['default']['six']
         assert p.lockfile['default']['six']['git'] == 'https://github.com/benjaminp/six.git'
         assert 'ref' in p.lockfile['default']['six']
+
+
+@pytest.mark.install
+@pytest.mark.index
+def test_install_named_index_alias(PipenvInstance, pypi):
+    # This uses the real PyPI since we need Internet to access the Git
+    # dependency anyway.
+    with PipenvInstance(pypi=pypi) as p:
+        with open(p.pipfile_path, 'w') as f:
+            contents = """
+[[source]]
+url = "{0}"
+verify_ssl = false
+name = "testindex"
+
+[[source]]
+url = "https://pypi.python.org/simple"
+verify_ssl = "true"
+name = "pypi"
+
+[packages]
+pytz = "*"
+six = {{version = "*", index = "pypi"}}
+
+[dev-packages]
+            """.format(os.environ.get('PIPENV_TEST_INDEX')).strip()
+            f.write(contents)
+        c = p.pipenv('install pytz --index pypi')
+        assert c.return_code == 0
```

Comment:
A few problems:

* Comment seems wrong. A lot of tests below have the wrong comment, it seems? None of them are using a Git repo afaict.
* This is not future-proof. If we add pytz to the mock PyPI (we might) this cannot detect an incorrect `--index` implementation.
* Probably don’t want to use `dict.get`. If `PIPENV_TEST_INDEX` is not set this needs to fail loudly.

Reply:
oh yeah I think I've been copy pasting a bunch, I can do some comment cleanup

- I agree about the package, we can use something that's only available on the test index like the one I am using in the other test